### PR TITLE
Tweaks to Boehm multithreaded API

### PR DESCRIPTION
**Pull request by jacob-hughes**
These are some minor changes that need fixing before this can be used properly in rustgc. Mostly because I misunderstood the way the Boehm API works the first time around.

### CODE REVIEW

File: allocator/src/lib.rs [hunk 1]
```diff
@@ -148,21 +148,25 @@ impl GcAllocator {
...
 
     pub fn init() {
-        unsafe { boehm::GC_start_performance_measurement() };
+        unsafe { boehm::GC_init() }
     }
 
     /// Returns true if thread was successfully registered.
     pub unsafe fn register_thread(stack_base: *mut u8) -> bool {
-        boehm::GC_register_my_thread(stack_base) != 0
+        return boehm::GC_register_my_thread(stack_base) == 0;
```
Comment (jacob-hughes) [inline allocator/src/lib.rs]: Damn, just spotted this. Fixed: b72774218cac691f39c1c2f0666e004d9d82d504

**Comment by ltratt** [comment]
Please squash.

**Comment by jacob-hughes** [comment]
Squashed

**Pull request merged**